### ISSUE TITLE
learning rate doesn't decrease

### ISSUE THREAD

**Issue by kaishijeng** [zylo117/Yet-Another-EfficientDet-Pytorch#22]
It seems learning rate doesn't decrease as shown in the following graph of the tensorboard:

![image](https://user-images.githubusercontent.com/12689630/79039249-d6f55500-7b94-11ea-8fbe-feb7efa93a66.png)

I have seen this behavior on every training in my system.
Do you know why?

**Comment by zylo117** [comment]
why would learning rate be dropping?

**Comment by kaishijeng** [comment]
Learning rate should decrease after each epoch so learning rate becomes smaller when training finds good weights of NN model. Isn't this a typical training practice?

**Comment by kaishijeng** [comment]
I have used other framework to train a network and below is learning rate graph in that training:
![image](https://user-images.githubusercontent.com/12689630/79039597-1b81f000-7b97-11ea-8dd9-25caf548bd25.png)

**Comment by zylo117** [comment]
> I have used other framework to train a network and below is learning rate graph in that training:
> ![image](https://user-images.githubusercontent.com/12689630/79039597-1b81f000-7b97-11ea-8dd9-25caf548bd25.png)

That's lr scheduler, you need to apply it yourself. Because you will never know when should you drop your learning rate, what if the dropping is too early, the convergence would be so slow. what if the dropping is too late, then overfitting. You can't be sure the timing and amount yourself, let alone the auto scheduling.

This impl has a reduce on plateau policy, it should be enough.
